src/game/movement.rs

```diff
@@ -1,5 +1,12 @@
-//! Handle player input and translate it into movement.
-//! Note that the approach used here is simple for demonstration purposes.
+//! Handle player input and translate it into movement through a character controller.
+//! A character controller is the collection of systems that govern the movement of characters.
+//!
+//! In our case, the character controller has the following logic:
+//! - Set [`MovementController`] directional intent based on directional keyboard input.
+//! - Apply movement based [`MovementController`] directional intent and maximum speed.
```

Comment:
```suggestion
//! - Set [`MovementController`] intent based on directional keyboard input.
//! - Apply movement based on [`MovementController`] intent and maximum speed.
```
"Directional keyboard input" here means like arrow keys / WASD.